vars/createGithubIssue.groovy

```diff
@@ -0,0 +1,67 @@
+def call(Map args = [:]){
+    def failureMessages = args.message
+    List<String> failedComponents = []
+
+    if (failureMessages.size() == 1 && failureMessages[0] == "Build failed") {
+        println("No component failed, skip creating github issue.")
+    }
+    else {
+        for(message in failureMessages.unique()){
+            java.util.regex.Matcher match = (message =~ /(?<=\bcomponent\s).*/)
+            String matched = match[0]
+            println(matched.split(" ")[0].trim())
+            failedComponents.add(matched.split(" ")[0].trim())
+        }
+
+        def yamlFile = readYaml(file: "manifests/${INPUT_MANIFEST}")
+        def currentVersion = yamlFile.build.version
+
+        for(component in yamlFile.components){
+            if (failedComponents.contains(component.name)) {
```

Comment:
The input manifest is going to contain the component, that is why it is building it right? I dont think you need to check that again here.

Reply:
Here I am using the same manifest that was passed and iterting over each component and checking if that component exists in the failed component list. If it exists i am passing the failed component repository info which I can only get from manifest file.

Reply:
Instead of iterating over all the components why not just use the mapped values. The failed components are going to be in the manifest thats not optional. However you can just get the values like repository url, etc directly from those mapped values. 
@prudhvigodithi WDYT?

Reply:
The `components` in the manifest yaml file is a list of map object, it's structure is like `[{name:component1,repository:<url1>, ref:<ref1>, checks:[<list>] }, {name:component2,repository:<url2>, ref:<ref2>, checks:[<list>] }........]`. Without iterating over the list not sure how I will find at which index the failed component value is present. @gaiksaya

Reply:
Got it. Didn't realize its a list.